### PR TITLE
Added isclose and allclose functions

### LINKED ISSUE #379: implement is close operator [FEA]
we should implement the isclose operator for comparing tensors.

https://www.w3schools.com/python/ref_math_isclose.asp

### PR DESCRIPTION
**Pull request by cliffburdick**
Closes #379

### CODE REVIEW

File: include/matx/operators/isclose.h [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        using matxop = bool;
+        using scalar_type = typename remove_cvref_t<Op2>::scalar_type;
+        using inner_type = typename inner_op_type_t<scalar_type>::type;
+
+        __MATX_INLINE__ std::string str() const { return "isclose()"; }
+
+        __MATX_INLINE__ IsCloseOp(Op1 op1, Op2 op2, double rtol, double atol) : 
+          op1_(op1), op2_(op2), rtol_(static_cast<inner_type>(rtol)), atol_(static_cast<inner_type>(atol)) 
+        {
+          static_assert(op1.Rank() == op2.Rank(), "Operator ranks must match in isclose()");
```
Comment (luitjens) [inline include/matx/operators/isclose.h:57]: Can you use  ASSERT_COMPATIBLE_OP_SIZES(op1);   ASSERT_COMPATIBLE_OP_SIZES(op2); here?

Comment (luitjens) [reply include/matx/operators/isclose.h:57]: Is there any reason the sizes must be equal?  This seems to prohibit broadcasting.   For example I could see someone doing something like this  isclose(op, 1).  ASSERT_COMPATIBILE_OP_SIZES is what you want I think.

File: include/matx/operators/isclose.h [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+          }
+
+        static __MATX_INLINE__ constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank()
+        {
+          return remove_cvref_t<Op1>::Rank();
+        }
+
+        constexpr __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t Size(int dim) const
+        {
+          return op1_.Size(dim);
```
Comment (luitjens) [inline include/matx/operators/isclose.h:79]: In Binary ops we do size like the following (should we match?):

      constexpr __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ auto Size(int dim) const noexcept
      {
        index_t size1 = detail::get_expanded_size<Rank()>(op1_, dim);
        index_t size2 = detail::get_expanded_size<Rank()>(op2_, dim);
        return detail::matx_max(size1,size2);
      }

File: include/matx/operators/isclose.h [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+        template <typename... Is>
+          __MATX_INLINE__ __MATX_DEVICE__ __MATX_HOST__ int operator()([[maybe_unused]] Is... indices) const 
+          {
+
+            return static_cast<int>(detail::_internal_abs(op1_(indices...) - op2_(indices...)) <= 
+               static_cast<inner_type>(atol_) + static_cast<inner_type>(rtol_) * detail::_internal_abs(op2_(indices...)));
+          }
+
+        static __MATX_INLINE__ constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank()
```
Comment (luitjens) [inline include/matx/operators/isclose.h:72]: In Binary operators we get rank like the following (should we match?):

      static __MATX_INLINE__ constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank()
      {
        return detail::matx_max(detail::get_rank<I1>(), detail::get_rank<I2>());
      }

File: include/matx/operators/isclose.h [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        {
+          static_assert(op1.Rank() == op2.Rank(), "Operator ranks must match in isclose()");
+          for (int32_t i = 0; i < op2.Rank(); i++) {
+            MATX_ASSERT_STR(op1.Size(i) == op2.Size(i), matxInvalidDim, 
+                "Size of each dimension must match in isclose()");
+          }
+        }
+
+        template <typename... Is>
+          __MATX_INLINE__ __MATX_DEVICE__ __MATX_HOST__ int operator()([[maybe_unused]] Is... indices) const
```
Comment (luitjens) [inline include/matx/operators/isclose.h:65]: int or bool for return type?  I'm not sure it matters.

**Review by luitjens — CHANGES_REQUESTED** [review]

File: include/matx/operators/isclose.h (hunk 1 above)
Comment (cliffburdick) [reply include/matx/operators/isclose.h:57]: Removed the restriction

**Pull request merged**